modules/tracking/src/gtrTracker.cpp

```diff
@@ -0,0 +1,203 @@
+			{
+				importer = dnn::createCaffeImporter(modelTxt, modelBin);
+			}
+			catch (const cv::Exception &err)        //Importer can throw errors, we will catch them
+			{
+				std::cerr << err.msg << std::endl;
+			}
+			if (!importer)
+			{
+				std::cerr << "Can't load network by using the following files: " << std::endl;
+				std::cerr << "prototxt:   " << modelTxt << std::endl;
+				std::cerr << "caffemodel: " << modelBin << std::endl;
+				std::cerr << "bvlc_googlenet.caffemodel can be downloaded here:" << std::endl;
+				std::cerr << "http://dl.caffe.berkeleyvision.org/bvlc_googlenet.caffemodel" << std::endl;
+				exit(-1);
+			}
+
+			importer->populateNet(net);
+			importer.release();                     //We don't need importer anymore
+
+			return true;
+		}
+
+		bool TrackerGOTURNImpl::updateImpl(const Mat& image, Rect2d& boundingBox)
+		{
+			int INPUT_SIZE = 227;
+			//Using prevFrame & prevBB from model and curFrame GOTURN calculating curBB
+			Mat curFrame = image.clone();
+			Mat prevFrame = ((TrackerGOTURNModel*)static_cast<TrackerModel*>(model))->getImage();
+			Rect2d prevBB = ((TrackerGOTURNModel*)static_cast<TrackerModel*>(model))->getBoundingBox();
+			Rect2d curBB;
+
+			float padTargetPatch = 2.0;
+			Rect2f searchPatchRect, targetPatchRect;
+			Point2f currCenter, prevCenter;
+			Mat prevFramePadded, curFramePadded;
+			Mat searchPatch, targetPatch;
+
+			prevCenter.x = (float)(prevBB.x + prevBB.width / 2);
+			prevCenter.y = (float)(prevBB.y + prevBB.height / 2);
+
+			targetPatchRect.width = (float)(prevBB.width*padTargetPatch);
+			targetPatchRect.height = (float)(prevBB.height*padTargetPatch);
+			targetPatchRect.x = (float)(prevCenter.x - prevBB.width*padTargetPatch / 2.0 + targetPatchRect.width);
+			targetPatchRect.y = (float)(prevCenter.y - prevBB.height*padTargetPatch / 2.0 + targetPatchRect.height);
+
+			copyMakeBorder(prevFrame, prevFramePadded, (int)targetPatchRect.height, (int)targetPatchRect.height, (int)targetPatchRect.width, (int)targetPatchRect.width, BORDER_REPLICATE);
+			targetPatch = prevFramePadded(targetPatchRect).clone();
+
+			copyMakeBorder(curFrame, curFramePadded, (int)targetPatchRect.height, (int)targetPatchRect.height, (int)targetPatchRect.width, (int)targetPatchRect.width, BORDER_REPLICATE);
+			searchPatch = curFramePadded(targetPatchRect).clone();
+
+			resize(targetPatch, targetPatch, Size(INPUT_SIZE, INPUT_SIZE));
+			resize(searchPatch, searchPatch, Size(INPUT_SIZE, INPUT_SIZE));
+
+			//Preprocess
+			//Resize
+			resize(targetPatch, targetPatch, Size(INPUT_SIZE, INPUT_SIZE));
+			resize(searchPatch, searchPatch, Size(INPUT_SIZE, INPUT_SIZE));
+
+			//Mean Subtract
+			targetPatch = targetPatch - 128;
+			searchPatch = searchPatch - 128;
+
+			//Convert to Float type
+			targetPatch.convertTo(targetPatch, CV_32FC1);
+			searchPatch.convertTo(searchPatch, CV_32FC1);
+
+			dnn::Blob targetBlob = dnn::Blob(targetPatch);
+			dnn::Blob searchBlob = dnn::Blob(searchPatch);
```

Comment:
I tried to use Blob::fromImages() during development, but couldn't find it's implementation, though it present in OpenCV documentation. Strangely enough, my code with constructors works, and dims() of created blobs are (1, 3, 227, 227). Most probably assertion fails because of type, it is defined as CV_32FC1 (168-169) - but assertion expect type CV_32F or CV_64F. I will fix it in next commit